app/components/shared/input/template.hbs

```diff
@@ -0,0 +1,9 @@
+<input
+  type={{this.chceckInputType}}
+  class='form-control {{this.size}}'
+  id={{@id}}
+  placeholder='{{@placeholder}}'
+  value={{@vale}}
+  {{on 'input' @onChange}}
```

Comment:
How about create action inside this component, and pass more comfortable `value` attribute, instead every time taking `event.target.value`. 
Example:
```suggestion
  {{on 'input' this.onChange}}
```

```js
// shared/input/component.js
@action
onChange(event) {
  const { value } = event.target;
  this.args.onChange?.(value, event);
}
```

Then when you'll using this shared component you'll have lovely parameters:
```hbs
<Shared::Input 
  @value={{this.value}}
  @onChange={{this.onChange}}
/>
```

```js
@action
onChange(value, /* event */) {
  this.value = value; // If you want to use event, you have it in second argument 🚀
}
```